RecoLocalTracker/SiPixelRecHits/src/PixelCPEClusterRepair.cc

```diff
@@ -0,0 +1,628 @@
+      int maxPixelCol = theClusterParam.theCluster->maxPixelCol();
+      int maxPixelRow = theClusterParam.theCluster->maxPixelRow();
+      int minPixelCol = theClusterParam.theCluster->minPixelCol();
+      int minPixelRow = theClusterParam.theCluster->minPixelRow();
+      
+      //--- Are we near either of the edges?
+      bool edgex = ( theDetParam.theRecTopol->isItEdgePixelInX( minPixelRow ) || theDetParam.theRecTopol->isItEdgePixelInX( maxPixelRow ) );
+      bool edgey = ( theDetParam.theRecTopol->isItEdgePixelInY( minPixelCol ) || theDetParam.theRecTopol->isItEdgePixelInY( maxPixelCol ) );
+      
+
+      //--- Check status of both template calls.
+      if ( (theClusterParam.ierr !=0) || (theClusterParam.ierr2 !=0) )   
+      {
+         // If reconstruction fails the hit position is calculated from cluster center of gravity
+         // corrected in x by average Lorentz drift. Assign huge errors.
+         //xerr = 10.0 * (float)theClusterParam.theCluster->sizeX() * xerr;
+         //yerr = 10.0 * (float)theClusterParam.theCluster->sizeX() * yerr;
+         
+         if(!GeomDetEnumerators::isTrackerPixel(theDetParam.thePart))
+            throw cms::Exception("PixelCPEClusterRepair::localPosition :")
+            << "A non-pixel detector type in here?";
+         
+         // Assign better errors based on the residuals for failed template cases
+         if ( GeomDetEnumerators::isBarrel(theDetParam.thePart) )
+         {
+            xerr = 55.0f * micronsToCm;
+            yerr = 36.0f * micronsToCm;
+         }
+         else
+         {
+            xerr = 42.0f * micronsToCm;
+            yerr = 39.0f * micronsToCm;
+         }
+         
+         //cout << "xerr = " << xerr << endl;
+         //cout << "yerr = " << yerr << endl;
+         
+         //return LocalError(xerr*xerr, 0, yerr*yerr);
+      }
+      else if ( theClusterParam.edgeTypeX_ || theClusterParam.edgeTypeY_ )
+      {
+         // for edge pixels assign errors according to observed residual RMS
+         if      ( theClusterParam.edgeTypeX_ && !theClusterParam.edgeTypeY_ )
+         {
+            xerr = 23.0f * micronsToCm;
+            yerr = 39.0f * micronsToCm;
+         }
+         else if ( !theClusterParam.edgeTypeX_ && theClusterParam.edgeTypeY_ )
+         {
+            xerr = 24.0f * micronsToCm;
+            yerr = 96.0f * micronsToCm;
+         }
+         else if ( theClusterParam.edgeTypeX_ && theClusterParam.edgeTypeY_ )
+         {
+            xerr = 31.0f * micronsToCm;
+            yerr = 90.0f * micronsToCm;
+         }
+      }
+      else 
+      {
+         xerr = theClusterParam.templSigmaX_ * micronsToCm;
+         yerr = theClusterParam.templSigmaY_ * micronsToCm;
+                  
+         // &&& should also check ierr (saved as class variable) and return
+         // &&& nonsense (another class static) if the template fit failed.
+      }       
+      
+      //cout << "xerr = " << xerr << endl;
```

Comment:
Please remove the commented out lines, or replace with some LogDebug or similar

Reply:
OK, if you require this change now.